trie/utils/verkle.go

```diff
@@ -262,16 +267,25 @@ func EvaluateAddressPoint(address []byte) *verkle.Point {
 }
 
 func GetTreeKeyStorageSlotWithEvaluatedAddress(evaluated *verkle.Point, storageKey *uint256.Int) []byte {
-	pos := storageKey.Clone()
+	// Note that `pos` must be a big.Int and not a uint256.Int, because the subsequent
+	// arithmetics operations could overflow. (e.g: imagine if storageKey is 2^256-1)
+	pos := storageKey.ToBig()
 	if storageKey.Cmp(codeStorageDelta) < 0 {
-		pos.Add(HeaderStorageOffset, storageKey)
+		pos.Add(headerStorageOffsetBig, pos)
 	} else {
-		pos.Add(MainStorageOffset, storageKey)
+		pos.Add(mainStorageOffsetBig, pos)
+	}
+	treeIndex, overflow := uint256.FromBig(pos.Div(pos, verkleNodeWidthBig))
+	if overflow { // Must never happen considering the EIP definition.
+		panic("tree index overflow")
 	}
-	treeIndex := new(uint256.Int).Div(pos, VerkleNodeWidth)
 	// calculate the sub_index, i.e. the index in the stem tree.
 	// Because the modulus is 256, it's the last byte of treeIndex
-	subIndexMod := new(uint256.Int).Mod(pos, VerkleNodeWidth)
+	subIndexMod, overflow := uint256.FromBig(pos.Mod(pos, verkleNodeWidthBig))
```

Comment:
I'm thinking we can definitely get rid of the `FromBig` (and the potential panic) here. It's just extracting the last byte. It was meant to be endinness-independent, but this might be pointless here because a `big.Int` is always big-endian, unlike `uint256.Int`.

Reply:
Done.